apps/root/src/hooks/earn/useEarnClaimDelayedWithdrawAction.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { DelayedWithdrawalPositions, EarnClaimDelayedWithdrawTypeData, TransactionTypes } from 'common-types';
+import { FormattedMessage } from 'react-intl';
+import useTrackEvent from '@hooks/useTrackEvent';
+import { shouldTrackError } from '@common/utils/errors';
+import useErrorService from '@hooks/useErrorService';
+import { Typography } from 'ui-library';
+import useTransactionModal from '@hooks/useTransactionModal';
+import { useTransactionAdder } from '@state/transactions/hooks';
+import useEarnService from '@hooks/earn/useEarnService';
+
+const useEarnClaimDelayedWithdrawAction = () => {
+  const trackEvent = useTrackEvent();
+  const errorService = useErrorService();
+  const earnService = useEarnService();
+  const addTransaction = useTransactionAdder();
+  const [, setModalLoading, setModalError, setModalClosed] = useTransactionModal();
+
+  const onClaimDelayedWithdraw = React.useCallback(
+    async (position: DelayedWithdrawalPositions) => {
+      const strategy = position.strategy;
+
+      // NOTE: For now, strategies should only have one claimable token
+      const claimToken = position.delayed.find((delayed) => delayed.ready.amount > 0n);
+
+      try {
+        if (!claimToken) throw new Error('No claimable token found');
+
+        setModalLoading({
+          content: (
+            <Typography variant="bodyRegular">
+              <FormattedMessage
+                description="earn.strategy-management.claim-delayed-withdraw.modal.loading"
+                defaultMessage="You are now claiming delayed withdraw from {farm}. Time to enjoy the rewards you've cultivated 🎉"
+                values={{ farm: strategy.farm.name }}
```

Comment:
lets also add the token symbol here and the amount in case the copy team want to use it